comp/metadata/inventoryagent/inventoryagentimpl/inventoryagent.go

```diff
@@ -256,6 +256,9 @@ func (ia *inventoryagent) Set(name string, value interface{}) {
 }
 
 func (ia *inventoryagent) getPayload() marshaler.JSONMarshaler {
+	ia.m.Lock()
+	defer ia.m.Unlock()
```

Comment:
Could we add some test around these changes?

Reply:
As discussed, testing is not worth adding a dedicated blocking mechanism in the code to test concurrent access.